cpp/common/src/codingstandards/cpp/EncapsulatingFunctions.qll

```diff
@@ -18,6 +18,39 @@ class MainFunction extends MainLikeFunction {
   }
 }
 
+/**
+ * A test function from the GoogleTest infrastructure.
+ *
+ * Such functions can be treated as valid EntryPoint functions during analysis
+ * of "called" or "unused" functions. It is not straightforward to identify
+ * such functions, however, they have certain features that can be used for
+ * identification. This can be refined based on experiments/real-world use.
+ */
+class GTestFunction extends MainLikeFunction {
```

Comment:
I would have a small preference for calling this `GoogleTestFunction` - it's more verbose, but easier to understand what it covers if you see it in code.